test/integration/analytical_policy.hpp

```diff
@@ -73,8 +73,8 @@ using yields = std::pair<micm::Species, double>;
 
 using SparseMatrixTest = micm::SparseMatrix<double>;
 
-template<class BuilderPolicy>
-void test_analytical_troe(BuilderPolicy& builder)
+template<class BuilderPolicy, class PrepareFunc, class PostpareFunc>
+void test_analytical_troe(BuilderPolicy& builder, PrepareFunc prepare_for_solve, PostpareFunc postpare_for_solve)
```

Comment:
If we end up keeping the functions, can they be arguments instead of templates??
```suggestion
template<class BuilderPolicy>
void test_analytical_troe(BuilderPolicy& builder, auto prepare_for_solve = [](auto& state) { },, auto  postpare_for_solve = [](auto& state) { })
```

Reply:
I tried to use arguments but it did not compile. Thus I assume that we have to use templates here.